lib/source-code/source-code.js

```diff
@@ -921,6 +971,84 @@ class SourceCode extends TokenStore {
         return configNodes;
     }
 
+    /**
+     * Returns an all directive nodes that enable or disable rules along with any problems
+     * encountered while parsing the directives.
+     * @returns {{problems:Array<Problem>,directives:Array<Directive>}} Information
+     *      that ESLint needs to further process the directives.
+     */
+    getDisableDirectives() {
+
+        // check the cache first
+        const cachedDirectives = this[caches].get("disableDirectives");
+
+        if (cachedDirectives) {
+            return cachedDirectives;
+        }
+
+        const problems = [];
+        const directives = [];
+
+        this.getInlineConfigNodes().forEach(comment => {
+            const { directivePart, justificationPart } = commentParser.extractDirectiveComment(comment.value);
+
+            // Step 1: Extract the directive text
+            const match = directivesPattern.exec(directivePart);
+
+            if (!match) {
+                return;
+            }
+
+            const directiveText = match[1];
+
+            // Step 2: Extract the directive value
+            const lineCommentSupported = /^eslint-disable-(next-)?line$/u.test(directiveText);
+
+            if (comment.type === "Line" && !lineCommentSupported) {
+                return;
+            }
+
+            // Step 3: Validate the directive does not span multiple lines
+            if (directiveText === "eslint-disable-line" && comment.loc.start.line !== comment.loc.end.line) {
+                const message = `${directiveText} comment should not span multiple lines.`;
+
+                problems.push({
+                    ruleId: null,
+                    message,
+                    loc: comment.loc
+                });
+                return;
+            }
+
+            // Step 4: Extract the directive value and create the Directive object
+            const directiveValue = directivePart.slice(match.index + directiveText.length);
+
+            switch (directiveText) {
+                case "eslint-disable":
+                case "eslint-enable":
+                case "eslint-disable-next-line":
+                case "eslint-disable-line": {
+                    const directiveType = directiveText.slice("eslint-".length);
+
+                    directives.push(new Directive({
+                        type: directiveType,
+                        node: comment,
+                        value: directiveValue,
```

Comment:
This seems different from what the RFC was proposing. Here, SourceCode returns a string `value` that is later parsed in the Linter as a comma-separated list of rule names.

Per the RFC, `SourceCode#getDisableDirectives()` should be doing this parsing, allowing for different formats in different languages. 

```js
interface SourceCode {
    //...
        getDisableDirectives(): Array<DisableDirective>;
    //...
}

interface DisableDirective {
    type: "enable" | "disable";
    ruleId: string;  // <--

    /**
     * The point from which the directive should be applied.
     */
    from: Location;

    /**
     * Location of the comment.
     */
    loc: LocationRange;
}
```

Is the new design intentional?

Reply:
Yes, I did deviate from the RFC here, just forgot to mention that part of the deviation in the original comment.

My thinking here is that if we limit the types of disable directives to the ones we already support, then `applyDisableDirectives()` will just work as-is for all languages and it dramatically reduces the complexity of implementing this feature for other languages.